server/src/test/java/org/elasticsearch/index/mapper/BooleanFieldMapperTests.java

```diff
@@ -130,26 +132,176 @@ public void testSerialization() throws IOException {
-        String mapping = Strings.toString(XContentFactory.jsonBuilder()
+    public void test_null() throws Exception{
+        this.parseAndCheck(FIELD1_IGNORE_MALFORMED, null, null);
+    }
+
+    public void test_true() throws Exception{
+        this.parseAndCheck(FIELD1_IGNORE_MALFORMED,true, BooleanFieldMapper.TRUE);
+    }
+
+    public void test_false() throws Exception{
+        this.parseAndCheck(FIELD1_IGNORE_MALFORMED,false, BooleanFieldMapper.FALSE);
+    }
+
+    public void test_on() throws Exception{
+        this.parseFails("on");
+    }
+
+    public void test_off() throws Exception{
+        this.parseFails("off");
+    }
+
+    public void test_yes() throws Exception{
+        this.parseFails("yes");
+    }
+
+    public void test_no() throws Exception{
+        this.parseFails("no");
+    }
+
+    public void test_0() throws Exception{
+        this.parseFails("0");
+    }
+
+    public void test_1() throws Exception{
+        this.parseFails("1");
+    }
+
+    public void test_string_IgnoreMalformed() throws Exception{
+        this.parseAndCheck(FIELD1_IGNORE_MALFORMED,
+            sourceWithString(),
+            null, BooleanFieldMapper.TRUE);
+    }
+
+    public void test_string() throws Exception{
+        this.parseFails(sourceWithString());
+    }
+
+    private BytesReference sourceWithString() throws Exception{
+        return BytesReference.bytes(XContentFactory.jsonBuilder()
             .startObject()
-                .startObject("type")
-                    .startObject("properties")
-                        .startObject("field")
-                            .field("type", "boolean")
-                        .endObject()
-                    .endObject()
-                .endObject()
+            .field(FIELD1, "**Never convertable to boolean**") // IGNORED
+            .field(FIELD2, true)
             .endObject());
-        DocumentMapper defaultMapper = parser.parse("type", new CompressedXContent(mapping));
-        BytesReference source = BytesReference.bytes(XContentFactory.jsonBuilder()
-                .startObject()
-                    // omit "false"/"true" here as they should still be parsed correctly
-                    .field("field", randomFrom("off", "no", "0", "on", "yes", "1"))
-                .endObject());
+    }
+
+    public void test_number_IgnoreMalformed() throws Exception{
+        this.parseAndCheck(FIELD1_IGNORE_MALFORMED,
+            this.sourceWithNumber(1),
+            null, BooleanFieldMapper.TRUE);
+    }
+
+    public void test_number() throws Exception{
+        this.parseFails(this.sourceWithNumber(1));
+    }
+
+    private BytesReference sourceWithNumber(final int value) throws Exception{
+        return BytesReference.bytes(XContentFactory.jsonBuilder()
+            .startObject()
+            .field(FIELD1, value) // IGNORED
+            .field(FIELD2, true)
+            .endObject());
+    }
+
+    public void test_object_IgnoreMalformed() throws Exception{
```

Comment:
Although the code is clear, the level of indirection here makes it hard for the reader to figure out that this (and similar other) test does. As a suggestion, what about combining test_object/test_object_IgnoreMalformed, then the code from `sourceWithObject` can be inlined in the test case. Also I would make the assertion on field1 and field2 explicit, even if that means a few more lines of code. In this case I would trade repetition for readability.